File: frontend/demo/component/upload/upload-button-theme-variant.ts
import 'Frontend/demo/init'; // hidden-source-line
import './upload-demo-helpers'; // hidden-source-line
import '@vaadin/button';
import '@vaadin/upload';
import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import { Notification } from '@vaadin/notification';
import type { UploadFileRejectEvent, UploadMaxFilesReachedChangedEvent } from '@vaadin/upload';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('upload-button-theme-variant')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  @state()
  private maxFilesReached = false;

  private uploadI18n = {
    dropFiles: {
      one: 'Drop PDF here',
    },
  };

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-upload
        max-files="1"
        accept="application/pdf,.pdf"
        .i18n="${this.uploadI18n}"
        @file-reject="${this.fileRejectHandler}"
        @max-files-reached-changed="${(event: UploadMaxFilesReachedChangedEvent) => {
          this.maxFilesReached = event.detail.value;
        }}"
      >
        <vaadin-button slot="add-button" theme="primary" ?disabled="${this.maxFilesReached}">
          Upload PDF...
        </vaadin-button>
      </vaadin-upload>
      <!-- end::snippet[] -->
    `;
  }

  fileRejectHandler(event: UploadFileRejectEvent) {
    Notification.show(`Error: ${event.detail.error} '${event.detail.file.name}'`);
  }
}

File: src/main/java/com/vaadin/demo/component/upload/UploadButtonThemeVariant.java
package com.vaadin.demo.component.upload;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.button.ButtonVariant;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.upload.Upload;
import com.vaadin.flow.component.upload.receivers.MemoryBuffer;
import com.vaadin.flow.router.Route;

@Route("upload-button-theme-variant")
public class UploadButtonThemeVariant extends Div {

    public UploadButtonThemeVariant() {
        MemoryBuffer buffer = new MemoryBuffer();
        // tag::snippet[]
        Upload upload = new Upload(buffer);
        upload.setAcceptedFileTypes("application/pdf", ".pdf");

        Button uploadButton = new Button("Upload PDF...");
        uploadButton.addThemeVariants(ButtonVariant.LUMO_PRIMARY);

        upload.setUploadButton(uploadButton);

        // Disable the upload button after the file is selected
        // Re-enable the upload button after the file is cleared
        upload.getElement().addEventListener("max-files-reached-changed", event -> {
            boolean maxFilesReached = event.getEventData().getBoolean("event.detail.value");
            uploadButton.setEnabled(!maxFilesReached);
        }).addEventData("event.detail.value");
        // end::snippet[]

        UploadExamplesI18N i18N = new UploadExamplesI18N();
        i18N.getDropFiles().setOne("Drop PDF here");
        upload.setI18n(i18N);

        add(upload);
    }

    public static class Exporter // hidden-source-line
            extends DemoExporter<UploadButtonThemeVariant> { // hidden-source-line
    } // hidden-source-line
}
